### PR TITLE
sysusers: support `u username -:300` style syntax

### PR DESCRIPTION
**Pull request by mvo5**
This PR implements the first part of RFE #8046. I.e. this allows to
write:
```
u username -:300
```
Where the uid is chosen automatcially but the gid is fixed.

### CODE REVIEW

File: src/sysusers/sysusers.c [hunk 1]
```diff
@@ -1591,11 +1591,12 @@ static int parse_line(const char *fname, unsigned line, const char *buffer) {
                                         i->id_set_strict = true;
                                         free_and_replace(resolved_id, uid);
                                 }
-                                r = parse_uid(resolved_id, &i->uid);
-                                if (r < 0)
-                                        return log_error_errno(r, "Failed to parse UID: '%s': %m", id);
-
-                                i->uid_set = true;
+                                if(!streq(resolved_id, "-")) {
```
Comment (keszybz) [inline src/sysusers/sysusers.c]: space after `if` missing.

**Comment by yuwata** [comment]
Thanks. I do not know we should support the following, but I guess it works. Could you add a test for it?
```
u user1 300
u user2 -:300
```

**Comment by mvo5** [comment]
added the extra test

**Comment by yuwata** [comment]
Thanks. lgtm. (The commit title lacks '`'.)

**Comment by yuwata** [comment]
I've forgot to say this. Please update the man page.

**Comment by mvo5** [comment]
Indeed, thank you! I updaetd the man-page and rebased on top of master.

File: src/sysusers/sysusers.c (hunk 1 above)
Comment (mvo5) [reply src/sysusers/sysusers.c]: Fixed! Sorry for that.

**Pull request merged**